zerver/tests/test_retention.py

```diff
@@ -324,3 +328,37 @@ def test_archiving_attachments_with_removing(self):
                 'messages__id', flat=True)),
             sorted(msgs_ids.values()))
 
+    def test_delete_archived_data(self):
+        # type: () -> None
+        msgs_ids = self._send_msgs_with_attachments()
+        exp_msgs_ids_dict = self._make_expired_messages()
+        archive_messages()
+
+        self.assertEqual(
+            ArchiveMessage.objects.count(),
+            len(exp_msgs_ids_dict['zulip_msgs_ids'] + exp_msgs_ids_dict['mit_msgs_ids']) + 1)
+        self.assertEqual(
+            ArchiveAttachment.objects.count(),
+            3
+        )
+        self._add_expired_date_to_archive_data()
+
+        delete_archived_data()
+        self.assertEqual(
+            ArchiveAttachment.objects.count(),
+            3
+        )
+        self.assertEqual(
+            ArchiveMessage.objects.count(),
+            0)
+        self._change_msgs_pub_date([msgs_ids['actual_message_id']],
+                                   timezone.now() - timedelta(days=101))
+        archive_messages()
+        self._add_expired_date_to_archive_data()
+        with mock.patch('zerver.lib.upload.LocalUploadBackend.delete_message_image',
+                        return_value=True):
+            delete_archived_data()
+        self.assertEqual(
+            ArchiveAttachment.objects.count(),
+            0
+        )
```

Comment:
It is checked do not deleting archive attachments if exists some  messages(not archived) which is  linked with this archive attachments. And deleting them if all related messages are archived and can be removed.